### PR TITLE
v0.6.0 -> v0.7.0 upgrade guide

### PR DESCRIPTION
**Pull request by iszubok**
An upgrade guide from v0.6.0 to v0.7.0
Covers both systemd and docker upgrade paths

### CODE REVIEW

File: testnet/validator upgrade 0.6.0 - 0.7.0.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+This guide assumes you have followed the instructions to crete a systemd service or docker container validator node v0.6.0 and you perform the upgrade on a machine running v0.6.0
+
+It is highly recommended to make the backup of your datadir after you stop v0.6.0 and before you start v0.7.0. Unless you changed your datadir, it should be located in `~/.laconicd`
+
+## Systemd service
+Skip this section if you use docker
+
+This is very similar to building v0.6.0. We keep v 0.6.0 running until v0.7.0 is built and only after the successful build we should replace 0.6.0 binary with v0.7.0. This is to avoid jailing your validator for long downtime.
+The general upgrade plan is the following:
+  1. Install Go v1.19 (we used 1.18 for v0.6.0)
```
Comment (sibuser) [inline testnet/validator upgrade 0.6.0 - 0.7.0.md:12]: I would pin to a specific version like 1.19.4

File: testnet/validator upgrade 0.6.0 - 0.7.0.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  2. Remove old copy of the github repository and build directory
+  3. Download the latest laconicd repository and checkout v0.7.0
+  4. Build laconicd binary (but not install in this moment)
+  5. Stop laconicd systemd service
+  6. Install recently built new version of laconicd
+  7. Start laconicd service
+
+You have ~10 minutes after step 5 to complete steps 6 and 7 before your validator is jailed for downtime. Getting jailed for downtime is not a disaster, however would require manual unjailing.
```
Comment (sibuser) [inline testnet/validator upgrade 0.6.0 - 0.7.0.md:20]: Would mark it as warning. People might oversee this part

File: testnet/validator upgrade 0.6.0 - 0.7.0.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+---
+
+### Stop `laconicd` systemd service
+
+```sh
+sudo systemctl stop laconicd
+```
+
+---
```
Comment (sibuser) [inline testnet/validator upgrade 0.6.0 - 0.7.0.md:88]: Probably good to verify that service is stopped

File: testnet/validator upgrade 0.6.0 - 0.7.0.md (hunk 1 above)
Comment (iszubok) [reply testnet/validator upgrade 0.6.0 - 0.7.0.md:12]: It is 1.19.5 later in the documentation

File: testnet/validator upgrade 0.6.0 - 0.7.0.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+### Remove old copy of `laconicd` build
+
+```sh
+# Remove the previous build directory
+cd ~
+rm -rf laconicd
```
[the remark this answers is not in the document]
Comment (iszubok) [reply testnet/validator upgrade 0.6.0 - 0.7.0.md:56]: No, this is where you clone the repository and build laconicd. .laconicd is the datadir which must be kept

File: testnet/validator upgrade 0.6.0 - 0.7.0.md (hunk 2 above)
Comment (iszubok) [reply testnet/validator upgrade 0.6.0 - 0.7.0.md:20]: agree

File: testnet/validator upgrade 0.6.0 - 0.7.0.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+## Docker container
+
+Upgrade plan:
+  1. Stop running v0.6.0 container
+  2. Delete v0.6.0 container
+  3. Create v0.7.0 container
+  4. Start v0.7.0 container
+
+
+### Stop running v0.6.0 container
```
[the remark this answers is not in the document]
Comment (sibuser) [reply testnet/validator upgrade 0.6.0 - 0.7.0.md:139]: Thanks!

**Review by i-norden — APPROVED** [review]
LGTM, I don't have much to add. Thanks!

**Pull request merged**